File: app/src/main/java/com/lusle/android/soon/Adapter/Holder/CompanyAlarmViewHolder.java
package com.lusle.android.soon.Adapter.Holder;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.google.android.material.switchmaterial.SwitchMaterial;
import com.lusle.android.soon.R;

public class CompanyAlarmViewHolder extends RecyclerView.ViewHolder {

    public ImageView companyLogo;
    public TextView companyName;
    public SwitchMaterial companyAlarmActive;

    public CompanyAlarmViewHolder(@NonNull View itemView) {
        super(itemView);

        companyLogo = itemView.findViewById(R.id.logo);
        companyName = itemView.findViewById(R.id.company_name);
        companyAlarmActive = itemView.findViewById(R.id.company_alarm_active);
        itemView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if(companyAlarmActive.isEnabled())
                    companyAlarmActive.setChecked(!companyAlarmActive.isChecked());
            }
        });
    }
}

File: app/src/main/java/com/lusle/android/soon/adapter/holder/CompanyAlarmViewHolder.kt
package com.lusle.android.soon.adapter.holder

import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.switchmaterial.SwitchMaterial
import com.lusle.android.soon.model.schema.Company
import com.lusle.android.soon.R
import com.lusle.android.soon.util.CircleTransform
import com.lusle.android.soon.adapter.CompanyAlarmSettingsAdapter
import com.lusle.android.soon.adapter.listener.OnCheckedChangeListener
import com.squareup.picasso.Picasso

class CompanyAlarmViewHolder(itemView: View, private val onCheckedChangeListener:OnCheckedChangeListener) : RecyclerView.ViewHolder(itemView) {
    private var companyLogo: ImageView = itemView.findViewById(R.id.logo)
    private var companyName: TextView = itemView.findViewById(R.id.company_name)
    private var companyAlarmActive: SwitchMaterial = itemView.findViewById(R.id.company_alarm_active)

    fun bind(company:Company){
        itemView.setOnClickListener {
            if (companyAlarmActive.isEnabled) companyAlarmActive.isChecked =
                !companyAlarmActive.isChecked
        }

        Picasso
            .get()
            .load("https://image.tmdb.org/t/p/w500" + company.logo_path)
            .centerCrop()
            .transform(CircleTransform())
            .fit()
            .error(R.drawable.ic_broken_image)
            .into(companyLogo)
        companyName.text = company.name
        try {
            companyAlarmActive.isEnabled = true
            companyAlarmActive.isChecked = false
            for (topic in (bindingAdapter as CompanyAlarmSettingsAdapter).topics) {
                if (topic.trim { it <= ' ' } == company.id.toString()) {
                    companyAlarmActive.isChecked = true
                }
            }
        } catch (e: Exception) {
            e.printStackTrace()
            companyAlarmActive.isEnabled = false
        }
        companyAlarmActive.setOnCheckedChangeListener { view, isChecked ->
            onCheckedChangeListener.onCheckedChangeListener(view, isChecked, company)
        }
    }
}
